contracts/contracts/RequestManager.sol

```diff
@@ -764,42 +833,64 @@ contract RequestManager is Ownable, LpWhitelist, RestrictedCalls, Pausable {
+        uint256 ethInToken
     ) external onlyOwner {
-        require(lpFeePPM <= 999_999, "Maximum PPM of 999999 exceeded");
-        require(protocolFeePPM <= 999_999, "Maximum PPM of 999999 exceeded");
-
         Token storage token = tokens[tokenAddress];
         token.transferLimit = transferLimit;
-        token.minLpFee = minLpFee;
-        token.lpFeePPM = lpFeePPM;
-        token.protocolFeePPM = protocolFeePPM;
-
-        emit TokenUpdated(
-            tokenAddress,
-            transferLimit,
-            minLpFee,
-            lpFeePPM,
-            protocolFeePPM
-        );
+        token.ethInToken = ethInToken;
+
+        emit TokenUpdated(tokenAddress, transferLimit, ethInToken);
     }
 
-    /// Set the finality period for the given target chain.
+    /// Update chain information for a given chain ID.
     ///
     /// .. note:: This function can only be called by the contract owner.
     ///
-    /// @param targetChainId The target chain ID.
-    /// @param finalityPeriod Finality period in seconds.
-    function setFinalityPeriod(
-        uint256 targetChainId,
-        uint256 finalityPeriod
+    /// @param chainId The chain ID of the chain.
+    /// @param finalityPeriod The finality period of the chain.
+    /// @param transferCost The transfer cost (fill, claim, withdraw) on the chain in WEI.
+    /// @param targetWeightPPM The share of the target chain costs (fill) in parts per million.
+    function updateChain(
+        uint256 chainId,
+        uint256 finalityPeriod,
+        uint256 transferCost,
+        uint256 targetWeightPPM
     ) external onlyOwner {
         require(finalityPeriod > 0, "Finality period must be greater than 0");
-        finalityPeriods[targetChainId] = finalityPeriod;
+        require(targetWeightPPM <= 999_999, "Maximum PPM of 999999 exceeded");
+
+        Chain storage chain = chains[chainId];
+        chain.finalityPeriod = finalityPeriod;
+        chain.transferCost = transferCost;
+        chain.targetWeightPPM = targetWeightPPM;
+
+        emit ChainUpdated(
+            chainId,
+            finalityPeriod,
+            transferCost,
+            targetWeightPPM
+        );
+    }
+
+    /// Update fees
+    ///
+    /// .. note:: This function can only be called by the contract owner.
+    ///
+    /// @param _minFeePPM Margin which is going to be applied to the minLpFee
+    /// @param _lpFeePPM LP percentage fee applied on transfer amount denominated in parts per million
+    /// @param _protocolFeePPM Protocol fee applied on transfer amount denominated in parts per million
+    /// @param _protocolFeePPM Protocol fee applied on transfer amount denominated in parts per million
+    function updateFees(
+        uint32 _minFeePPM,
+        uint32 _lpFeePPM,
+        uint32 _protocolFeePPM
+    ) external onlyOwner {
+        require(_lpFeePPM <= 999_999, "Maximum PPM of 999999 exceeded");
+        require(_protocolFeePPM <= 999_999, "Maximum PPM of 999999 exceeded");
```

Comment:
We're missing another `require` in the same way for `minFeePPM`.

Reply:
I left it open on purpose. The reason for it is that it is very unlikely that we will apply a percentage fee higher than 100%. But as for the minFee I see situations where we want to apply a fee greater than 100%. Not sure that it will ever happen, but naturally the minFeePPM is much larger than lpFeePPM and protocolFeePPM just to begin with.